gvfs-helper.c

```diff
@@ -0,0 +1,2424 @@
+			      "failed to start HTTP request");
+	} else {
+		run_active_slot(slot);
+		if (params->b_write_to_file)
+			fflush(params->tempfile->fp);
+		gh__response_status__set_from_slot(status, slot);
+	}
+
+	if (params->progress)
+		stop_progress(&params->progress);
+
+	gh__response_status__dump(params, status);
+	trace2_region_leave("gvfs-helper", params->label.buf, NULL);
+}
+
+static int option_parse_product_mode(const struct option *opt,
+				     const char *arg, int unset)
+{
+	if (unset) /* should not happen */
+		return error(_("missing value for switch '%s'"),
+			     opt->long_name);
+
+	else if (!strcmp(arg, "auto"))
+		gh__cmd_opts.product_mode = GH__PRODUCT_MODE__AUTO;
+
+	else if (!strcmp(arg, "vfs"))
+		gh__cmd_opts.product_mode = GH__PRODUCT_MODE__VFS;
+
+	else if (!strcmp(arg, "scalar"))
+		gh__cmd_opts.product_mode = GH__PRODUCT_MODE__SCALAR;
+
+	else if (!strcmp(arg, "none"))
+		gh__cmd_opts.product_mode = GH__PRODUCT_MODE__NONE;
+
+	else
+		return error(_("invalid value for switch '%s'"),
+			     opt->long_name);
+
+	return 0;
+}
+
+static int option_parse_cache_server_mode(const struct option *opt,
+					  const char *arg, int unset)
+{
+	if (unset) /* should not happen */
+		return error(_("missing value for switch '%s'"),
+			     opt->long_name);
+
+	else if (!strcmp(arg, "verify"))
+		gh__cmd_opts.cache_server_mode =
+			GH__CACHE_SERVER_MODE__VERIFY_DISABLE;
+
+	else if (!strcmp(arg, "error"))
+		gh__cmd_opts.cache_server_mode =
+			GH__CACHE_SERVER_MODE__VERIFY_ERROR;
+
+	else if (!strcmp(arg, "disable"))
+		gh__cmd_opts.cache_server_mode =
+			GH__CACHE_SERVER_MODE__DISABLE;
+
+	else if (!strcmp(arg, "allow"))
+		gh__cmd_opts.cache_server_mode =
+			GH__CACHE_SERVER_MODE__ALLOW_WITHOUT_VERIFY;
+
+	else
+		return error(_("invalid value for switch '%s'"),
+			     opt->long_name);
+
+	return 0;
+}
+
+/*
+ * The config is processed before we have parsed the command
+ * line args so we don't yet know which (if any) cache-server
+ * to use.  Remember both URLs and decide later.
+ *
+ * Also, look for well-known core config settings and capture
+ * values.  We do this directly so that we don't have to know
+ * about the existence of global variables from gvfs.c (or
+ * later, scalar.c).
+ *
+ * Also, break from the traditional config-helper model and
+ * do not intercept these values, but rather allow normal
+ * config processing continue.
+ */
+static int config_cb(const char *k, const char *v, void *data)
+{
+	if (!strcmp(k, "gvfs.cache-server")) {
+		const char *v2 = NULL;
+
+		if (!git_config_string(&v2, k, v) && v2 && *v2)
+			gh__cmd_opts.vfs_url = transport_anonymize_url(v2);
+		free((char *)v2);
+	}
+
+	else if (!strcmp(k, "scalar.cache-server")) {
+		const char *v2 = NULL;
+		if (!git_config_string(&v2, k, v) && v2 && *v2)
+			gh__cmd_opts.scalar_url = transport_anonymize_url(v2);
+		free((char *)v2);
+	}
+
+	else if (!strcmp(k, "core.gvfs"))
+		gh__cmd_opts.is_core_vfs = 1;
+
+	else if (!strcmp(k, "core.scalar"))
```

Comment:
Currently Scalar is setting `core.gvfs` (with the specific flags it wants) and not `core.scalar` is that going to cause any issues?

Reply:
No.  I was anticipating that we'd get a core.scalar flag as part of the global rename.  If that's not the case, I can get rid of this.